feat(profile-metadata): store profile metadata in warp

ashneverdawn opened the pull request:
### What this PR does 📖

- profile metadata (integrations) used to just be locally stored in the browser. It is now stored in warp. This allows us to access a friend's metadata in subsequent PRs.
- this also makes some changes to allow handling of arbitrary key/value pairs in the profile metadata
- also displays the user's metadata from warp in their profile card (accessible from chat)

### Which issue(s) this PR fixes 🔨

### Special notes for reviewers 🗒️

### Additional comments 🎤

luisecm commented:
The failing tests on CI are because of the settings profile snapshots tests used to validate changing profile banner/picture are finding a difference between the base snapshot not having this new part for Accounts. I will create a PR in the testing repo to update snapshots and I can merge once this PR is merge too

![image](https://github.com/user-attachments/assets/3a8d71c0-6d91-4794-a81d-0ad034556064)

ashneverdawn commented on src/lib/components/ui/IntegrationDisplay.svelte:
```diff
@@ -2,28 +2,44 @@
+    export let editable: boolean = false
 
-    let display = INTEGRATIONS[integration.kind].display
+    function toIntegrationKind(key: string) {
```
done

ashneverdawn commented on src/lib/components/ui/IntegrationDisplay.svelte:
```diff
@@ -2,28 +2,44 @@
+        }
+        return integration_kind
+    }
+    function display(key: string) {
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -139,51 +140,49 @@
-            const updatedIntegrations = [...currentIntegrations, { ...selectedIntegration }]
-            Store.setIntegrations(updatedIntegrations)
-            selectedIntegration = { kind: Integrations.Generic, location: "", meta: "" }
+        if (selectedKey !== "" && selectedKeyEditValue !== "") {
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -386,63 +385,76 @@
+                                <Input alt bind:value={selectedKey} disabled={user.integrations.has(selectedKey)} />
+                            {/if}
+                        </div>
+                        <img class="integration-logo" src="/assets/brand/{toIntegrationKind(selectedKey)}.png" alt="Platform Logo" />
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -386,63 +385,76 @@
+                </div>
+
+                {#if $showEditIntegrations}
+                    <Label text={user.integrations.has(selectedKey) ? "Edit Integration" : "Add New"} />
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -386,63 +385,76 @@
+                            <Input alt bind:value={selectedKeyEditValue} />
+                        </div>
+
+                        <Button text={user.integrations.has(selectedKey) ? "Save" : "Add"} on:click={user.integrations.has(selectedKey) ? saveEditedIntegration : addIntegration}>
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -386,63 +385,76 @@
+                        </Button>
                         <Button
-                            text="Add"
+                            text="Cancel"
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -139,51 +140,49 @@
-        Store.setIntegrations(currentIntegrations)
+    function removeIntegration(key: string) {
+        MultipassStoreInstance.removeMetadata(key)
+        showEditIntegrations.set(false)
```
done

ashneverdawn commented on src/routes/settings/profile/+page.svelte:
```diff
@@ -90,6 +90,7 @@
     })
 
     let user: User = get(Store.state.user)
+    Store.state.user.subscribe(u => (user = u))
```
done

ashneverdawn commented:
> Hey Hugo! Noticed some weird behavior with the profile pictures while testing
> 
> So when the user is updating their username or status once they hit enter or click save the custom profile picture disappears. When this happens it also doesnt go back to showing the default Uplink image it will just appear blank (so in the screen recording you can see that the picture will appear blank but I tested it again while writing this and after I updated my profile picture the default image did re-appear so not sure what thats about). It also doesnt seem like any strange errors are happening during this action.

I just tried it and I was able to reproduce that on my end. Also, sett banner seems to disappear profile pic and vice versa. Not sure what's causing it. Seems unrelated to what I added in this PR, but I could be missing something.
